### PR TITLE
Added support for Imenu

### PR DESCRIPTION
**Pull request by atimchenko92**
According to Issue #42.
I am quite new to Elisp and Lisp in general, so further corrections are welcomed! :smile:

### CODE REVIEW

File: solidity-mode.el [hunk 1]
```diff
@@ -493,6 +493,24 @@ Cursor must be at the function's name.  Does not currently work for constructors
   (interactive)
   (solidity--start-gasestimate (thing-at-point 'symbol 'no-properties)))
 
+;;; Support for imenu
```
Comment (LefterisJP) [inline solidity-mode.el]: Hey @atimchenko92. Thank you very much. My colleague @rakanalh who uses Imenu tested it and it works.

I have one request only. Can you add a changelog entry [here](https://github.com/ethereum/emacs-solidity/blob/d6c48a1cb64d3c8a825dc0d06c839f2cacd4d289/changelog.MD#nightly)? Something like:
"Added support for Imenu".

Comment (atimchenko92) [reply solidity-mode.el]: Added entry to changelog as well. My pleasure! :)

**Comment by LefterisJP** [comment]
Hey @atimchenko92 can you rebase the last merge commit away? It should not be needed.

**Comment by atimchenko92** [comment]
Now it should be good

**Comment by LefterisJP** [comment]
@atimchenko92 yep! Thank you. Much appreciated.

**Pull request merged**